compiler/src/iree/compiler/Codegen/Common/TileSwizzle.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/TileSwizzle.h"
+#include "llvm/ADT/STLExtras.h"
+
+namespace mlir::iree_compiler {
+
+llvm::raw_ostream &operator<<(llvm::raw_ostream &os,
+                              TileSwizzle::Dim::Kind kind) {
+  switch (kind) {
+  case TileSwizzle::Dim::Kind::Internal:
+    return os << "Internal";
+  case TileSwizzle::Dim::Kind::CrossThread:
+    return os << "CrossThread";
+  case TileSwizzle::Dim::Kind::CrossIntrinsic:
+    return os << "CrossIntrinsic";
+  }
+}
```

Comment:
This broke the gcc build: https://github.com/iree-org/iree/actions/runs/11030265747/job/30634464951#step:4:6937

```
[6307/8097] Building CXX object compiler/src/iree/compiler/Codegen/Common/CMakeFiles/iree_compiler_Codegen_Common_Common.objects.dir/TileSwizzle.cpp.o
FAILED: compiler/src/iree/compiler/Codegen/Common/CMakeFiles/iree_compiler_Codegen_Common_Common.objects.dir/TileSwizzle.cpp.o 
/usr/local/bin/ccache /usr/bin/g++-9  -I/__w/iree/iree -I/__w/iree/iree/build-gcc -I/__w/iree/iree/third_party/llvm-project/llvm/include -I/__w/iree/iree/build-gcc/llvm-project/include -I/__w/iree/iree/third_party/llvm-project/mlir/include -I/__w/iree/iree/build-gcc/llvm-project/tools/mlir/include -I/__w/iree/iree/third_party/llvm-project/lld/include -I/__w/iree/iree/build-gcc/llvm-project/tools/lld/include -I/__w/iree/iree/compiler/src -I/__w/iree/iree/build-gcc/compiler/src -I/__w/iree/iree/runtime/src -I/__w/iree/iree/build-gcc/runtime/src -I/__w/iree/iree/build-gcc/runtime/src/iree/base/internal/flatcc -I/__w/iree/iree/llvm-external-projects/iree-dialects/include -I/__w/iree/iree/build-gcc/llvm-external-projects/mlir-iree-dialects/include -isystem /__w/iree/iree/third_party/flatcc/include -O3  -fPIC -fvisibility=hidden -fno-rtti -fno-exceptions -Wall -Werror -Wno-error=deprecated-declarations -Wno-address -Wno-address-of-packed-member -Wno-comment -Wno-format-zero-length -Wno-uninitialized -Wno-overloaded-virtual -Wno-invalid-offsetof -Wno-sign-compare -Wno-unused-function -Wno-unknown-pragmas -Wno-unused-but-set-variable -Wno-misleading-indentation -fmacro-prefix-map=/__w/iree/iree=iree -I/__w/iree/iree/third_party/flatcc/include/ -I/__w/iree/iree/third_party/flatcc/include/flatcc/reflection/ -std=gnu++17 -MD -MT compiler/src/iree/compiler/Codegen/Common/CMakeFiles/iree_compiler_Codegen_Common_Common.objects.dir/TileSwizzle.cpp.o -MF compiler/src/iree/compiler/Codegen/Common/CMakeFiles/iree_compiler_Codegen_Common_Common.objects.dir/TileSwizzle.cpp.o.d -o compiler/src/iree/compiler/Codegen/Common/CMakeFiles/iree_compiler_Codegen_Common_Common.objects.dir/TileSwizzle.cpp.o -c /__w/iree/iree/compiler/src/iree/compiler/Codegen/Common/TileSwizzle.cpp
/__w/iree/iree/compiler/src/iree/compiler/Codegen/Common/TileSwizzle.cpp: In function 'llvm::raw_ostream& mlir::iree_compiler::operator<<(llvm::raw_ostream&, mlir::iree_compiler::TileSwizzle::Dim::Kind)':
/__w/iree/iree/compiler/src/iree/compiler/Codegen/Common/TileSwizzle.cpp:22:1: error: control reaches end of non-void function [-Werror=return-type]
   22 | }
      | ^
cc1plus: all warnings being treated as errors
```

Could assert on the default / fallthrough case, or implement some logic in case an unhandled enum value gets passed in.

Reply:
Thanks for flagging! I have added a fix to https://github.com/iree-org/iree/pull/18547 which I'm about to merge.